src/prefs.nim

```diff
@@ -0,0 +1,116 @@
+import asyncdispatch, times, macros, tables, xmltree
+import types
+
+withCustomDb("prefs.db", "", "", ""):
+  try:
+    createTables()
+  except DbError:
+    discard
+
+type
+  PrefKind* = enum
+    checkbox, select, input
+
+  Pref* = object
+    name*: string
+    label*: string
+    case kind*: PrefKind
+    of checkbox:
+      defaultState*: bool
+    of select:
+      defaultOption*: string
+      options*: seq[string]
+    of input:
+      defaultInput*: string
+      placeholder*: string
+
+const prefList*: Table[string, seq[Pref]] = {
+  "Privacy": @[
+    Pref(kind: input, name: "replaceTwitter",
+         label: "Replace Twitter links with Nitter (blank to disable)",
+         defaultInput: "nitter.net", placeholder: "Nitter hostname"),
```

Comment:
This should be the hostname of the nitter instance instead

Reply:
It's a placeholder for the input field which is too small for that, I can change it to "Nitter instance"

Reply:
I do not mean the placeholder. I mean the `defaultInput` should be the hostname of the nitter instance (e.g. if selfhosting nitter)

Reply:
Ah, I tried doing that but it's not really possible besides patching the code. The simple reason for this is that this (immutable) list of preferences is created at compile-time to generate a bunch of code, and the config is read at runtime so it's too late. I can provide a compile-time configuration, so you can set it like this `nimble build -d:hostname=nitter.net`

Reply:
If this is not possible directly here, than what about adding a config file? A config file will probably be needed later anyways (for e.g. rss feeds?).

Reply:
Would be an option, yes.